manifests/image.pp

```diff
@@ -74,10 +75,17 @@
     $image_find    = "${docker_command} images | cut -d ' ' -f 1 | egrep '^(docker\\.io/)?${image}$'"
   }
 
+  if $build_args {
+    validate_array($build_args)
+    $build_args_string = join ( regsubst ($build_args, '^(.*)$', '--build-arg \1 '), '')
+  } else {
+    $build_args_string = ''
+  }
+
   if $docker_dir {
-    $image_install = "${docker_command} build -t ${image_arg} ${docker_dir}"
+    $image_install = "${docker_command} build ${build_args_string}-t ${image_arg} ${docker_dir}"
```

Comment:
I _think_ (off the top of my head) you might not need the `!empty($build_args)` check or the second clause. Worth a try.